redis/connection.py

```diff
@@ -360,6 +365,17 @@ def on_connect(self):
             if str_if_bytes(self.read_response()) != "OK":
                 raise ConnectionError("Error setting client name")
 
+        try:
+            # set the library name and version
+            if self.lib_name:
+                self.send_command("CLIENT", "SETINFO", "LIB-NAME", self.lib_name)
+                self.read_response()
+            if self.lib_version:
+                self.send_command("CLIENT", "SETINFO", "LIB-VER", self.lib_version)
```

Comment:
Pipeline SETINFO and SELECT same as in the async suggestion above.

Reply:
Specifically not sent because of the interface - but a suggestion we'd debated previously. At this level, the pipeline shouldn't be used (yet). IMHO once reorganizing connections, all items for connect should be pipelined, agreed.

Reply:
Well, you probably debated it out of band, not in this PR :)  What "interface" are you speaking of?

A pipeline isn't an object "to be used", it is just a way of sending a bunch of commands and not waiting for each individual response before sending the next command.  It is always supported by the server.
The spec specifically recommends pipelining startup commands to reduce latency.  We now have two extra roundtrips to the server at startup in addition to the optional one to select a db.